packages/footer/components/Newsletter.tsx

```diff
@@ -0,0 +1,239 @@
+}) => {
+   return (
+      <FormControl w="full" isInvalid={subscription.error != null}>
+         <Input
+            ref={inputRef}
+            type="email"
+            disabled={subscription.status !== SubscriptionStatus.Idle}
+            data-testid="newsletter-email-input"
+            variant="filled"
+            bg="white"
+            color="gray.900"
+            fontSize="sm"
+            _focus={{
+               bg: 'white',
+               boxShadow: 'outline',
+            }}
+            placeholder={placeholder}
+         />
+         <FormErrorMessage>{subscription.error}</FormErrorMessage>
+      </FormControl>
+   );
+};
+
+const NewsletterHeader = ({
+   title,
+   subtitle,
+}: {
+   title: string;
+   subtitle: string;
+}) => {
+   return (
+      <Box textAlign="left">
+         <Text fontWeight="bold" fontSize="lg" color="white" my={2}>
+            {title}
+         </Text>
+         <Text color="gray.300" fontSize="md" fontWeight="normal" mt={3}>
+            {subtitle}
+         </Text>
+      </Box>
+   );
+};
+
+const NewsletterForm = ({
+   inputRef,
+   placeholder,
+   buttonText,
+   isSubscribed,
+   subscription,
+   onSubmit,
+}: {
+   inputRef: React.RefObject<HTMLInputElement>;
+   placeholder: string;
+   buttonText: string;
+   isSubscribed: boolean;
+   subscription: Subscription;
+   onSubmit: (event: React.FormEvent<HTMLElement>) => Promise<void>;
+}) => {
+   if (isSubscribed) {
+      return null;
+   }
+   return (
+      <HStack
+         as="form"
+         onSubmit={onSubmit}
+         data-testid="footer-newsletter-form"
+         spacing="3"
+         w={{
+            base: 'full',
+            xl: 'auto',
+         }}
+         justify={{
+            base: 'center',
+            md: 'flex-end',
+         }}
+         align="flex-start"
+      >
+         <NewsletterInput
+            inputRef={inputRef}
+            placeholder={placeholder}
+            subscription={subscription}
+         />
+         <Button
+            type="submit"
+            data-testid="footer-newsletter-subscribe-button"
+            isLoading={subscription.status === SubscriptionStatus.Subscribing}
+            disabled={subscription.status !== SubscriptionStatus.Idle}
+            bg="blue.ifixit"
+            color="white"
+            border="none"
+            fontWeight="bold"
+            _hover={{ bg: 'blue.ifixit', boxShadow: '0 0 5px gray' }}
+         >
+            {buttonText}
+         </Button>
+      </HStack>
+   );
+};
+
+const NewsletterSubscribed = ({ isSubscribed }: { isSubscribed: boolean }) => {
+   if (!isSubscribed) {
+      return null;
+   }
+   return (
+      <Flex
+         height="42px"
+         flexDirection="row"
+         alignItems="center"
+         padding="12px"
+         gap="8px"
+         w="100%"
+         backgroundColor="#EDF6FF"
+         border="1px"
+         borderColor="#77B5FF"
+         borderStyle="solid"
+         borderRadius="4px"
```

Comment:
I'm a little surprised by all the explicit numbers and colors here, I would have expected to depend more on our chakra theme (and it's colors and spacing numbers).

Reply:
Good catch. This code was copied from iFixit/ifixit where they are not defined. I can switch them over.

Reply:
Colors: [2374ee0](https://github.com/iFixit/react-commerce/pull/1342/commits/2374ee0f899c5d43db0dd674a61c3e5e13dfce0d)
Spacing: [66f37ec](https://github.com/iFixit/react-commerce/pull/1342/commits/66f37ec5fbba6888204cf8593f6651ed5c7cabc8)